src/Confluent.Kafka/Consumer.cs

```diff
@@ -139,7 +129,26 @@ public bool Consume(out Message<TKey, TValue> message, int millisecondsTimeout)
                 return false;
             }
 
-            message = msg.Deserialize(KeyDeserializer, ValueDeserializer);
+            try
+            {
+                message = msg.Deserialize(KeyDeserializer, ValueDeserializer);
+            }
+            catch (KafkaException ex)
+            {
+                var erroredMsg = new Message(
+                    msg.Topic,
+                    msg.Partition,
+                    msg.Offset,
+                    msg.Key,
+                    msg.Value,
+                    msg.Timestamp,
+                    ex.Error
+                );
+                OnConsumeError(this, erroredMsg);
```

Comment:
good catch. it was this originally, i changed it when experimenting and forgot to change it back..